test/pouch_cli_images_test.go

```diff
@@ -1,57 +1,89 @@
 package main
 
 import (
-	"os/exec"
+	"strings"
 
+	"github.com/alibaba/pouch/apis/types"
+	"github.com/alibaba/pouch/client"
+	"github.com/alibaba/pouch/test/command"
+	"github.com/alibaba/pouch/test/environment"
 	"github.com/go-check/check"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/pkg/errors"
 )
 
 // PouchImagesSuite is the test suite fo help CLI.
-type PouchImagesSuite struct {
-}
+type PouchImagesSuite struct{}
 
 func init() {
 	check.Suite(&PouchImagesSuite{})
 }
 
 // SetUpSuite does common setup in the beginning of each test suite.
 func (suite *PouchImagesSuite) SetUpSuite(c *check.C) {
-	SkipIfFalse(c, IsLinux)
+	SkipIfFalse(c, environment.IsLinux)
 
-	// Pull test image
-	cmd := exec.Command("pouch", "pull", testImage)
-	cmd.Run()
+	command.PouchRun("pull", busyboxImage).Assert(c, icmd.Success)
 }
 
-// SetUpTest does common setup in the beginning of each test.
-func (suite *PouchImagesSuite) SetUpTest(c *check.C) {
-	// TODO
-}
+// TestImagesWorks tests "pouch image" work.
+func (suite *PouchImagesSuite) TestImagesWorks(c *check.C) {
+	image, err := getImageInfo(apiClient, busyboxImage)
+	c.Assert(err, check.IsNil)
 
-// TearDownSuite does cleanup work in the end of each test suite.
-func (suite *PouchImagesSuite) TearDownSuite(c *check.C) {
-	// TODO: Remove test image
-}
+	// without flag
+	{
+		res := command.PouchRun("images").Assert(c, icmd.Success)
+		items := imagesListToKV(res.Combined())[busyboxImage]
+
+		c.Assert(items[0], check.Equals, image.ID)
+	}
+
+	// with -q and --quiet
+	{
+		resQ := command.PouchRun("images", "-q").Assert(c, icmd.Success)
+		resQuiet := command.PouchRun("images", "--quiet").Assert(c, icmd.Success)
+
+		c.Assert(resQ.Combined(), check.Equals, resQuiet.Combined())
+		c.Assert(strings.TrimSpace(resQ.Combined()), check.Equals, image.ID)
+	}
 
-// TearDownTest does cleanup work in the end of each test.
-func (suite *PouchImagesSuite) TearDownTest(c *check.C) {
-	// TODO add cleanup work
+	// with --digest
+	{
+		res := command.PouchRun("images", "--digest").Assert(c, icmd.Success)
+		items := imagesListToKV(res.Combined())[busyboxImage]
+		c.Assert(items[2], check.Equals, image.Digest)
+	}
 }
 
-// TestImagesWorks tests "pouch image" work.
-func (suite *PouchImagesSuite) TestImagesWorks(c *check.C) {
+// imagesListToKV parse "pouch images" into key-value mapping.
+func imagesListToKV(list string) map[string][]string {
+	// skip header
+	lines := strings.Split(list, "\n")[1:]
+
+	res := make(map[string][]string)
+	for _, line := range lines {
+		if strings.TrimSpace(line) == "" {
+			continue
+		}
+
+		items := strings.Fields(line)
+		res[items[1]] = items
+	}
+	return res
+}
 
-	// TODO: nil input should return success
-	// TODO: add wrong args.
-	args := map[string]bool{
-		"": false,
+// getImageInfo is used to retrieve the information about image.
+func getImageInfo(apiClient client.ImageAPIClient, name string) (types.ImageInfo, error) {
```

Comment:
We need to add an API `GET /images/{idOrName}/json` 😄

Reply:
On my plan. Haha